qcodes_contrib_drivers/drivers/Attocube/ANC350.py

```diff
@@ -0,0 +1,511 @@
+        actuator: Selects the actuator to be used for the axis from actuator presets.
+        actuator_name: Get the name of the currently selected actuator
+        capacitance: Performs a measurement of the capacitance of the piezo motor and returns the result. If no
+            motor is connected, the result will be 0.
+            The function doesn't return before the measurement is complete; this will take a few seconds of time.
+    """
+
+    def __init__(self, parent: "ANC350", name: str, axis: int):
+        super().__init__(parent, name)
+
+        self._axis = axis
+
+        self.add_parameter("position",
+                           label="Position",
+                           get_cmd=self._get_position,
+                           unit="mm or m°")
+
+        self.add_parameter("frequency",
+                           label="Frequency",
+                           get_cmd=self._get_frequency,
+                           set_cmd=self._set_frequency,
+                           vals=vals.Ints(1, 5000),
+                           unit="Hz")
+
+        self.add_parameter("amplitude",
+                           label="Amplitude",
+                           get_cmd=self._get_amplitude,
+                           set_cmd=self._set_amplitude,
+                           vals=vals.Numbers(0, 70),
+                           unit="V")
+
+        self.add_parameter("status",
+                           label="Status",
+                           get_cmd=self._get_status)
+
+        self.add_parameter("target_position",
+                           label="Target Position",
+                           get_cmd=None,
+                           set_cmd=self._set_target_position,
+                           vals=vals.Numbers(),
+                           unit="mm or m°")
+
+        self.add_parameter("target_range",
+                           label="Target Range",
+                           get_cmd=None,
+                           set_cmd=self._set_target_range,
+                           vals=vals.Numbers(),
+                           unit="mm or m°")
+
+        self.add_parameter("actuator",
+                           label="Actuator",
+                           get_cmd=None,
+                           set_cmd=self._set_actuator,
+                           vals=vals.Ints(0, 255))
+
+        self.add_parameter("actuator_type",
+                           label="Actuator Type",
+                           get_cmd=self._get_actuator_type)
+
+        self.add_parameter("actuator_name",
+                           label="Actuator Name",
+                           get_cmd=self._get_actuator_name)
+
+        self.add_parameter("capacitance",
+                           label="Capacitance",
+                           get_cmd=self._get_capacitance,
+                           unit="nF")
+
+        if self._parent._version_no >= 4:
+            voltage_get: Optional[Callable] = self._get_voltage
```

Comment:
I think you forgot to run the function _get_voltage here. Shouldn't be: voltage_get: Optional[Callable] = self._get_voltage()

Reply:
No, the function-object is assigned here, not the function's return value.

Reply:
OK. Then , how is this function object useful in the get_cmd of the voltage parameter, below? I think the user expects to get the voltage value when types voltage() or voltage.get().

Reply:
The function-object is passed to `add_parameter` as argument `get_cmd`. When the user calls `voltage()` or `voltage.get()`, the call is redirected to that function-object. It's the same as with function `_set_voltage` which is passed as argument `set_cmd`.

The only difference is that `_get_voltage` is only supported by firmware version 4. With firmware version 3 there is no getter for voltage because the instrument doesn't support it. So, `voltage_get` (= `get_cmd`) was set to `None`, so that Qcodes returns the cached value (last setpoint), when calling the getter.

Reply:
My bad. It's right.